icechunk-python/src/errors.rs

```diff
@@ -71,12 +75,81 @@ impl From<SessionError> for PyIcechunkStoreError {
 impl From<PyIcechunkStoreError> for PyErr {
     fn from(error: PyIcechunkStoreError) -> Self {
         match error {
+            PyIcechunkStoreError::SessionError(SessionError::Conflict {
+                expected_parent,
+                actual_parent,
+            }) => ConflictError::new_err(PyConflictErrorData {
+                expected_parent: expected_parent.map(|s| s.to_string()),
+                actual_parent: actual_parent.map(|s| s.to_string()),
+            }),
+            PyIcechunkStoreError::SessionError(SessionError::RebaseFailed {
```

Comment:
We cant create exceptions with custom methods or properties from rust, so instead we pass a custom object to the args. Which prints a nice message, but unfortunately means the user needs to go to Error.args[0] to get the actual data properties. Not idea but it works fine. 

Should we repackage the errors with our own custom exceptions in the python world from here? We can just wrap it in `session.py`